react/modules/tasks/queries.md

```diff
@@ -2,25 +2,109 @@
-
-2. Required Query Features:
-   - Cache the results of API calls
-   - Display loading states while data is being fetched
-   - Handle error states appropriately
-   - Implement proper data invalidation and refetching strategies
-   - Add a manual cache invalidation control (e.g., a refresh button) that forces a new API call instead of using cached data
-   - All previous API calls from your application should be converted to use the chosen query solution
-
-3. Specific Requirements:
-   - Data loading states should be visible to users (loading indicators)
-   - Error states should be properly handled and displayed to users
-   - The cached data should persist between page navigations
+## Functional Requirements (max **100 points**)
+
+### Feature 1: API Integration (**25 points**)
+
+**As a** developer
+**I want** to use a modern query library for API calls
+**So that** my data fetching is efficient and maintainable
+
+**Scenario:** Query Library Setup
+
+- **Given** I have chosen a state management solution (Redux or Zustand)
+- **When** I set up API querying
+- **Then** I use RTK Query (for Redux) or TanStack Query (for Zustand) for all API calls
+
+**Acceptance Criteria:**
+
+- RTK Query or TanStack Query is integrated and configured. [10 points]
+- All API calls are made using the chosen query library. [10 points]
+- Proper cache invalidation strategies are implemented. [5 points]
+
+### Feature 2: Data Caching and Loading States (**25 points**)
+
+**As a** user
+**I want** data to be cached and see loading indicators
+**So that** navigation is fast and I know when data is loading
+
+**Scenario:** Data Caching and Loading
+
+- **Given** I navigate between pages or open item details
+- **When** data is being fetched
+- **Then** a loading indicator is displayed
+- **And** previously fetched data is cached and reused when possible
+
+**Acceptance Criteria:**
+
+- Loading indicators are shown while data is being fetched. [13 points]
+- Data is cached and reused between navigations. [12 points]
+
+### Feature 3: Error Handling (**10 points**)
+
+**As a** user
+**I want** to see clear error messages when API calls fail
+**So that** I understand what went wrong
+
+**Scenario:** Error State Display
+
+- **Given** an API call fails
+- **When** I am using the app
+- **Then** a clear, human-readable error message is displayed
+
+**Acceptance Criteria:**
+
+- Error states are handled and displayed to users. [10 points]
+
+### Feature 4: Manual Cache Invalidation (**10 points**)
+
+**As a** user
+**I want** to manually refresh data
+**So that** I can ensure I have the latest information
+
+**Scenario:** Manual Refresh
+
+- **Given** I am viewing a list or details
+- **When** I click a refresh button
+- **Then** the cache is invalidated and a new API call is made
+
+**Acceptance Criteria:**
+
+- Manual cache invalidation control (refresh button) works as expected. [10 points]
```

Comment:
Are we referring to the browser's refresh button or do we want students to implement one?